src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -419,32 +419,26 @@ private static boolean isInSwitchRule(DetailAST ast) {
     }
 
     /**
-     * Checks if current expression is a switch labeled expression. If so,
-     * braces are not allowed e.g.:
-     * <p>
-     * {@code
-     * case 1 -> 4;
+     * Checks if the current case or default label is a switch labeled expression.
+     * If so, braces are not allowed e.g.:
+     * <pre>
+     * return switch(x) {
+     *    case Integer i -&gt; 3; // no braces allowed
+     *    case String s -&gt; s.length(); // no braces allowed
+     *    default -&gt; {throw new Exception();} // braces are allowed
      * }
-     * </p>
+     * </pre>
      *
-     * @param ast the ast to check
-     * @return true if current expression is a switch labeled expression.
+     * @param ast the ast to check representing a case or default label.
+     * @return true if current label is a switch labeled expression.
      */
     private static boolean isSwitchLabeledExpression(DetailAST ast) {
-        final DetailAST parent = ast.getParent();
-        return switchRuleHasSingleExpression(parent);
-    }
-
-    /**
-     * Checks if current switch labeled expression contains only a single expression.
-     *
-     * @param switchRule {@link TokenTypes#SWITCH_RULE}.
-     * @return true if current switch rule has a single expression.
-     */
-    private static boolean switchRuleHasSingleExpression(DetailAST switchRule) {
-        final DetailAST possibleExpression = switchRule.findFirstToken(TokenTypes.EXPR);
-        return possibleExpression != null
-                && possibleExpression.getFirstChild().getFirstChild() == null;
+        final DetailAST caseParent = ast.getParent();
+        final DetailAST switchAST = caseParent.getParent();
```

Comment:
Hm, I figured this would give us some checkstyle violations, please do `switchAst`

Reply:
`AST` is approved by our config.
https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle-checks.xml#L819

Just look at our AST class if you need to know why we did that. Hint: `DetailAST`

Reply:
Better to use `Ast`.
Let's not grow bad wording, even it is allowed. 

We might need to suppress certain violations to let it be forbidden for new code. But this could be code out side of this PR.